src/main/_docs/integration-guide/integration-version-control.md

```diff
@@ -25,6 +25,50 @@ For other git-based or SVN-based repository providers, please refer to their doc
 ## GitLab oAuth
 Currently it's not possible for Codenvy to use oAuth integration with GitLab. Although GitLab supports oAuth for clone operations, pushes are not supported. You can track [this GitLab issue](https://gitlab.com/gitlab-org/gitlab-ce/issues/18106) in their issue management system.
 
+## BitBucket Server oAuth
+### Setup oAuth for BitBucket Server
+#### 1. Generate Keys
+SSH to your BitBucket Server instance and generate RSA key-pairs
+```shell
+# generate the key pairs
+$ openssl genrsa -out private.pem 1024
+
+# copy the key and pem to a location where it can be accessed
+$ openssl rsa -in private.pem -pubout > public.pub`
+$ openssl pkcs8 -topk8 -inform pem -outform pem -nocrypt -in private.pem -out privatepkcs8.pem 
+```
+#### 2. Setup Codenvy in BitBucket Server
+In the Bitbucket Server as an Admin:
+
+1. Go to Administration -> Application Links
+1. Enter your Codenvy URL in the 'application url' field and press the 'Create new link' button.
+1. When  the 'Configure Application URL' window appears press the 'Continue' button.
+1. In the ‘Link applications’ window fill in the following fields and press the 'Continue' button:
+  - Application Name : Codenvy
+  - Service Provider Name : Codenvy
+  - Consumer key : `generated by BBS` (Save this string, you will need it later)
```

Comment:
Could you please explain what does `generated by BBS` mean. This field will be empty and you should specify it manually.